Undefined symbols: _objc_msgSend

**Describe the bug**
FreeRDP 2.0.0 and 2.1.0 do not build on macOS 10.13 and earlier:

```
[ 92%] Linking C shared library libfreerdp-client2.dylib
cd /path/to/build/client/common && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/freerdp-client.dir/link.txt --verbose=ON
/usr/bin/clang -pipe -Os -DNDEBUG -I/opt/local/include -DWINPR_DLL -arch x86_64 -mmacosx-version-min=10.13 -dynamiclib -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl,-headerpad_max_install_names -compatibility_version 2.0.0 -current_version 2.1.0 -o libfreerdp-client2.2.1.0.dylib -install_name @rpath/libfreerdp-client2.2.dylib CMakeFiles/freerdp-client.dir/client.c.o CMakeFiles/freerdp-client.dir/cmdline.c.o CMakeFiles/freerdp-client.dir/compatibility.c.o CMakeFiles/freerdp-client.dir/file.c.o CMakeFiles/freerdp-client.dir/__/__/channels/client/tables.c.o CMakeFiles/freerdp-client.dir/__/__/channels/client/addin.c.o  -Wl,-rpath,/path/to/build/libfreerdp -Wl,-rpath,/path/to/build/winpr/libwinpr ../../channels/drdynvc/client/libdrdynvc-client.a ../../channels/remdesk/client/libremdesk-client.a ../../channels/rdpsnd/client/librdpsnd-client.a ../../channels/rdpdr/client/librdpdr-client.a ../../channels/rdp2tcp/client/librdp2tcp-client.a ../../channels/rail/client/librail-client.a ../../channels/encomsp/client/libencomsp-client.a ../../channels/cliprdr/client/libcliprdr-client.a ../../channels/video/client/libvideo-client.a ../../channels/urbdrc/client/liburbdrc-client.a ../../channels/rdpsnd/client/librdpsnd-client.a ../../channels/rdpgfx/client/librdpgfx-client.a ../../channels/rdpei/client/librdpei-client.a ../../channels/geometry/client/libgeometry-client.a ../../channels/echo/client/libecho-client.a ../../channels/disp/client/libdisp-client.a ../../channels/audin/client/libaudin-client.a ../../channels/smartcard/client/libsmartcard-client.a ../../channels/serial/client/libserial-client.a ../../channels/printer/client/libprinter-client.a ../../channels/parallel/client/libparallel-client.a ../../channels/drive/client/libdrive-client.a ../../channels/urbdrc/client/libusb/liburbdrc-client-libusb.a ../../channels/rdpsnd/client/mac/librdpsnd-client-mac.a ../../channels/rdpsnd/client/fake/librdpsnd-client-fake.a ../../channels/printer/client/cups/libprinter-client-cups.a ../../channels/audin/client/mac/libaudin-client-mac.a -framework CoreServices ../../channels/urbdrc/client/liburbdrc-client.a /opt/local/lib/libusb-1.0.dylib -framework AVFoundation -framework CoreFoundation /usr/lib/libcups.dylib ../../libfreerdp/libfreerdp2.2.1.0.dylib ../../winpr/libwinpr/libwinpr2.2.1.0.dylib -framework AudioToolbox -framework CoreAudio -framework ApplicationServices
Undefined symbols for architecture x86_64:
  "_objc_msgSend", referenced from:
      _rdpsnd_mac_open in librdpsnd-client-mac.a(rdpsnd_mac.m.o)
      _rdpsnd_mac_set_volume in librdpsnd-client-mac.a(rdpsnd_mac.m.o)
      _rdpsnd_mac_play in librdpsnd-client-mac.a(rdpsnd_mac.m.o)
      _rdpsnd_mac_close in librdpsnd-client-mac.a(rdpsnd_mac.m.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [client/common/libfreerdp-client2.2.1.0.dylib] Error 1
```

They do build on macOS 10.14 and 10.15.

**To Reproduce**
Steps to reproduce the behavior:
1. Build FreeRDP 2.0.0 or 2.1.0 on macOS 10.13 or earlier

**Expected behavior**
Successful build

**Application details**
* FreeRDP version: 2.0.0 and 2.1.0
* CMake version: 3.17.2
* Compiler: Apple LLVM version 9.1.0 (clang-902.0.39.2)

**Environment (please complete the following information):**
 - OS: macOS
 - Version/Distribution: High Sierra 10.13.6 
 - Architecture: x86_64

**Additional context**
The problem was reported to MacPorts [here](https://trac.macports.org/ticket/60344).

The problem was not present in 2.0.0-rc4.

It builds successfully if I add `-framework Cocoa` to LDFLAGS but I don't know why it works on 10.14 and 10.15 without doing that.

--- a/channels/rdpsnd/client/mac/CMakeLists.txt
+++ b/channels/rdpsnd/client/mac/CMakeLists.txt
@@ -19,6 +19,7 @@
 
 define_channel_client_subsystem("rdpsnd" "mac" "")
 
+find_library(COCOA_LIBRARY Cocoa REQUIRED)
 FIND_LIBRARY(CORE_FOUNDATION CoreFoundation)
 FIND_LIBRARY(CORE_AUDIO CoreAudio REQUIRED)
 FIND_LIBRARY(AUDIO_TOOL AudioToolbox REQUIRED)
@@ -38,6 +39,7 @@ set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS}
     ${AUDIO_TOOL}
     ${AV_FOUNDATION}
     ${CORE_AUDIO}
+    ${COCOA_LIBRARY}
     ${CORE_FOUNDATION})
 
 set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} freerdp winpr)

